Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/TestPages/ScreenshotConditionalApp.cs

```diff
@@ -0,0 +1,435 @@
+		{
+			_app.PinchToZoomIn(query, duration);
+		}
+
+		public void PinchToZoomIn(string marked, TimeSpan? duration = null)
+		{
+			_app.PinchToZoomIn(marked, duration);
+		}
+
+		public void PinchToZoomInCoordinates(float x, float y, TimeSpan? duration)
+		{
+			_app.PinchToZoomInCoordinates(x, y, duration);
+		}
+
+		public void PinchToZoomOut(Func<AppQuery, AppQuery> query, TimeSpan? duration = null)
+		{
+			_app.PinchToZoomOut(query, duration);
+		}
+
+		public void PinchToZoomOut(string marked, TimeSpan? duration = null)
+		{
+			_app.PinchToZoomOut(marked, duration);
+		}
+
+		public void PinchToZoomOutCoordinates(float x, float y, TimeSpan? duration)
+		{
+			_app.PinchToZoomOutCoordinates(x, y, duration);
+		}
+
+		public void WaitFor(Func<bool> predicate, string timeoutMessage = "Timed out waiting...", TimeSpan? timeout = null,
+			TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			_app.WaitFor(predicate, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public AppResult[] WaitForElement(Func<AppQuery, AppQuery> query, string timeoutMessage = "Timed out waiting for element...",
+			TimeSpan? timeout = null, TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			return _app.WaitForElement(query, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public AppResult[] WaitForElement(string marked, string timeoutMessage = "Timed out waiting for element...",
+			TimeSpan? timeout = null, TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			return _app.WaitForElement(marked, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public AppWebResult[] WaitForElement(Func<AppQuery, AppWebQuery> query, string timeoutMessage = "Timed out waiting for element...",
+			TimeSpan? timeout = null, TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			return _app.WaitForElement(query, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public void WaitForNoElement(Func<AppQuery, AppQuery> query, string timeoutMessage = "Timed out waiting for no element...",
+			TimeSpan? timeout = null, TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			_app.WaitForNoElement(query, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public void WaitForNoElement(string marked, string timeoutMessage = "Timed out waiting for no element...",
+			TimeSpan? timeout = null, TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			_app.WaitForNoElement(marked, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public void WaitForNoElement(Func<AppQuery, AppWebQuery> query, string timeoutMessage = "Timed out waiting for no element...",
+			TimeSpan? timeout = null, TimeSpan? retryFrequency = null, TimeSpan? postTimeout = null)
+		{
+			_app.WaitForNoElement(query, timeoutMessage, timeout, retryFrequency, postTimeout);
+		}
+
+		public FileInfo Screenshot(string title)
+		{
+#if SCREENSHOTS
```

Comment:
So the idea is that `SCREENSHOTS` is not defined by default and if you need to debug your UITest then you rebuild with it defined before you submit your test to test cloud, right?